influence_benchmark/vectorized_environment/vectorized_environment.py

```diff
@@ -2,71 +2,50 @@
 
 import yaml
 
+from influence_benchmark.backend.backend import Backend
 from influence_benchmark.environment.environment import Environment
 from influence_benchmark.environment.state import State
 from influence_benchmark.root import PROJECT_ROOT
+from influence_benchmark.utils.utils import load_yaml
 from influence_benchmark.vectorized_environment.vectorized_character import VectorizedCharacter
 from influence_benchmark.vectorized_environment.vectorized_preference_model import VectorizedPreferenceModel
 from influence_benchmark.vectorized_environment.vectorized_transition_model import VectorizedTransitionModel
 
 
 class VecEnv:
-    def __init__(
-        self,
-        env_configs: List[Dict],
-        PM_backend_model: str,
-        TM_backend_model: str,
-        char_backend_model: str,
-        device: str,
-    ):
+    def __init__(self, env_configs: List[Dict], backend: Backend):
         self.env_configs = env_configs
-        self.envs = [Environment({**config, "vectorized": True}) for config in env_configs]
-        self._validate_envs()
+        self.envs = [Environment({**config, "vectorized": True}, backend=backend) for config in env_configs]
+        # self._validate_envs()
 
-        self.TM_backend_model = TM_backend_model
-        self.PM_backend_model = PM_backend_model
-        self.char_backend_model = char_backend_model
-
-        openai_models = ["gpt-4o", "gpt-4-turbo", "gpt-3.5-turbo"]
-        self.TM_backend_type = "openai" if TM_backend_model in openai_models else "huggingface"
-        self.PM_backend_type = "openai" if PM_backend_model in openai_models else "huggingface"
-        self.char_backend_type = "openai" if char_backend_model in openai_models else "huggingface"
-
-        self.device = device
+        self.backend = backend
         self.setup_models()
 
-    def _validate_envs(self):
-        assert all(isinstance(env, Environment) for env in self.envs), "All elements must be Environment instances"
-        assert all(env.config == self.envs[0].config for env in self.envs), "All environments must have the same config"
+    # def _validate_envs(self):
+    #     assert all(isinstance(env, Environment) for env in self.envs), "All elements must be Environment instances"
+    #     assert all(env.config == self.envs[0].config for env in self.envs), "All environments must have the same config"
```

Comment:
Do we not need this anymore? Seems like an easy location for a bug

Reply:
I don't think we do?